plugin/openstudio/lib/dialogs/SurfaceMatchingDialog.rb

```diff
@@ -136,13 +136,32 @@ def intersect(selection)
 
       # iterate through spaces to create intersecting geometry
       spaces.each do |space|
-        entity = space.entity
-        entity.entities.intersect_with(true, entity.transformation, entity.entities.parent, entity.transformation, false, selection.to_a)
+		entity = space.entity
+		before_faces = entity.entities.grep(Sketchup::Face) #create array of faces before the intersect
+		entity.entities.intersect_with(true, entity.transformation, entity.entities.parent, entity.transformation, false, selection.to_a)
+        edges = entity.entities.grep(Sketchup::Edge)
+		edges.each {|edge| edge.find_faces} #Heal empty loops (no face) after intersection, this may not be necessary?
+
+		#Ski90Moo:This will find faces with AttributeDictionaries copied from the base face (this is causing the faces to be added to the drawing_interface but a new model surface object is not created because all the copies point to an existing surface object.) Rough work around solution is to delete the dictionary, delete the sketchup entity, and redraw it.  This prompts the downstream observer to create a new model surface object and add the sketchup entities to the drawing_interface.  There is probably a more direct solution without having to delete and redraw, but I do not fully understand how the observer works.
+		
+		after_faces = entity.entities.grep(Sketchup::Face)  #create array of faces after the intersect
+		new_faces = after_faces - before_faces  #find the new faces (that just copied the attributes from the base face)
+		temp = []  #create array to temporarily store new_faces vertices
+		new_faces.each do |face|
+			face.delete_attribute('OpenStudio')  #delete dictionary attributes from the copied face so it will not delete the OS model object when deleted
```

Comment:
The problem with this is that you lose all information associated with the OpenStudio object.  Also what deletes the OpenStudio object so we don't create new ones?

Reply:
I believe the ModelManager tracks and deletes these.  Are you getting some that are duplicated or left over?

I thought it would be a concern regarding the surface properties (ie. boundary conditions and sun/wind exposure) getting reset from the original undivided surface.  However, I did some testing and remarkably the surface properties carry forward.